pkg/machines/components/deleteResource.jsx

```diff
@@ -54,6 +49,12 @@ export class DeleteResource extends React.Component {
         this.setState({ showModal: false, dialogError: undefined });
     }
 
+    delete() {
+        this.props.deleteHandler()
+                .then(() => this.close())
```

Comment:
Also this is a bit problematic. if the disk if successfully deleted the parent component will be unmounted and finally this will be unmounted so we don't need to execute this specifically. Calling this.close explicitly might result in a race condition where react will complain about trying to update and unmounted component.

Reply:
Alright. The bug is mainly fixed by adding key value to DeleteResource component, so this.close() doesn't need to be there